### PR TITLE
Bring back essentials 2

### PR DESCRIPTION
**Pull request by BillWagner**
Bring back in the essentials content.

The first commit brings back earlier commit by @shoag, but restores the files accidentally deleted.

The second commit fixes a number of build warnings due to link issues.

Given the time, and size of this, we may want to wait to merge.

cc @shoag, @mairaw, @tdykstra, @richlander.

### CODE REVIEW

**Comment by stevehoag** [comment]
Thanks Bill! I agree, we should wait until things stabilize tomorrow.

File: docs/core/rid-catalog.md [hunk 1]
```diff
@@ -119,7 +119,7 @@ wish to check if new ones are added, please check back here.
     * `linuxmint.17.2-x64`
     * `linuxmint.17.3-x64`
 
-## OS X RIDs
+## macOS RIDs
```
Comment (richlander) [inline docs/core/rid-catalog.md]: This really is an OS X RID. The version change starts with Sierra. El Capitan and before are still OS X. It's a question of how we're going to show the two sets of RIDs once Sierra is out. I'd suggest that we list them as two separate categories.

File: docs/core/tools/dotnet-publish.md [hunk 2]
```diff
@@ -88,5 +88,5 @@ Publishes the current application using the `netcoreapp1.0` framework.
 	
 `dotnet publish --framework netcoreapp1.0 --runtime osx.10.11-x64`
     
-Publishes the current application using the `netcoreapp1.0` framework and runtime for `OS X 10.10`. This RID has to 
+Publishes the current application using the `netcoreapp1.0` framework and runtime for `macOS 10.10`. This RID has to
```
Comment (richlander) [inline docs/core/tools/dotnet-publish.md]: 10.10 and 10.11 are not macOS and we didn't rename the RIDs.

File: docs/standard/basetypes/commontypesystem.md [hunk 3]
```diff
@@ -0,0 +1,446 @@
+---
+title: Common Type System
```
Comment (richlander) [inline docs/standard/basetypes/commontypesystem.md]: Probably good to put dashes in that file name.

**Comment by richlander** [comment]
Several of the file could benefit from dashes in the filenames, "common-type-system.md"

File: docs/scenarios/solution-authoring/cli-golden-path.md [hunk 4]
```diff
@@ -22,7 +22,7 @@ The scripts in this document describe the steps necessary to build a number of t
 Prerequisites
 -------------
 
-These scripts assume that .NET Core RC2 CLI SDK preview and VS Code or another code editor are installed on the machine.
+These scripts assume that .NET Core 1.0 and VS Code or another code editor are installed on the machine.
```
Comment (svick) [inline docs/scenarios/solution-authoring/cli-golden-path.md]: Doesn't this (and the F# article) require the .NET Core SDK?

**Comment by tdykstra** [comment]
When I click the "Files changed" link I get "Sorry, we could not display the entire diff because it was too big." -- what then is the best way to review the changes in the PR?

**Comment by richlander** [comment]
Two options:

- Pull the branch locally.
- Use the Web UI "jump to file" feature.

File: docs/standard/basetypes/commontypesystem.md (hunk 3 above)
Comment (tdykstra) [inline docs/standard/basetypes/commontypesystem.md]: We already have a common-type-system.md file in the standard folder.  Maybe we should combine the content?

File: docs/standard/basetypes/conversio/conversiontables.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
+---
+title: Type Conversion Tables
```
Comment (tdykstra) [inline docs/standard/basetypes/conversio/conversiontables.md]: Why is this in a "conversio" folder - is that a typo for "conversion"?  Also there's only one file in the folder, so we probably don't need the folder.

File: docs/standard/basetypes/index.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+[Formatting Types](format/index.md) - Describes how to format strings using the string format specifiers.
+
+[Manipulating Strings](manipulating/index.md) - Describes how to manipulate and format strings.
+
+[Parsing Strings](parsing/index.md) - Describes how to convert strings into types.
+
+## Related Sections
+
+[Common Type System](commontypesystem.md) - Describes types used in .NET Core.
```
Comment (tdykstra) [inline docs/standard/basetypes/index.md]: need to remove refs to Core in these docs ("used in .NET Core")

File: docs/scenarios/solution-authoring/cli-golden-path.md (hunk 4 above)
Comment (cartermp) [inline docs/scenarios/solution-authoring/cli-golden-path.md]: Yes, these will require the SDK Preview 2 bits.

**Comment by BillWagner** [comment]
@stevehoag If you've picked all of this work back up, I'll close this PR.  Please let me know.

**Comment by stevehoag** [comment]
@BillWagner Yes, I'll handle it. Thanks!

**Pull request closed without merging**